include/native/parallel_execution_native.h

```diff
@@ -766,7 +765,7 @@ auto parallel_execution_native::map_reduce(
   constexpr sequential_execution seq;
   auto process_chunk = [&](auto f, std::size_t sz, std::size_t id) {
     partial_results[id] = seq.map_reduce(f, sz,
-        std::forward<Identity>(partial_results[id]), 
+        std::forward<Identity>(identity_),
```

Comment:
Why?

Reply:
The idea is that now every "child" map_reduce uses the same arguments as the parent.
This changes a bit the design. Let me know if you think we should only use the identity on the first map_reduce child and use the first element of every chunk as a sort of identity, the same way it is done in the last step of this algorithm to put in common all results form each parallized thread.

Reply:
I see your approach.
We need that this is double checked by @Drioastorga to make sure we are not forgetting any corner case.

Reply:
Note that this is related to issue #334

Reply:
Just a comment. This change is only valid if the identity value introduced is the identity value for the operator. Otherwise the value introduced as identity will be part of each partial result. For instance, if the combine operator is a plus operator, and the user inroduces "2" as identity value, these value will be added number_of_ threads times in the final result.

Therefore, we have two choices:
- To perform this modification and document this behaviour. Thus, only when the value is an actual identity value the results will be correct in parallel versions.
- To not modify and maintain the original behaviour.

Reply:
We have discussed this and we concluded that we SHOULD accept this change.

Reply:
Revert back to the original in this pull, except removing the name of the parameter. We should recover it once we fix this issue.